### DISCUSSION TITLE
How can I add a custom label or variable per machine?

### DISCUSSION THREAD

**Discussion by pcwa-ahendricks** [twpayne/chezmoi#3546] [Q&A]
I want to create a distinction between my "work computers", and my "personal computers", that way I can sync different files or different file contents based on that distinction. How can I add a custom label/tag or variable per machine? Any ideas would be appreciated. (i'm coming from yadm, where this functionality exists, to chezmoi)

**Comment by twpayne** [comment]
Use a boolean config variable for each tag. This is [what I do](https://github.com/twpayne/dotfiles/blob/32e220c0f7c44f36432ea7a8a33449014246b403/home/.chezmoi.toml.tmpl).

**Comment by halostatue** [comment]
Start from [Manage Machine to Machine Differences](https://www.chezmoi.io/user-guide/manage-machine-to-machine-differences/). From there, dig deeper into Chezmoi templating and a finally templated config file for data management (`.chezmoi.$FORMAT.tmpl`). You will be able to apply differences based on machine characteristics, on configuration data, or on other options.

**Comment by pcwa-ahendricks** [comment]
Thank you all for the feedback and for pointing me in the right direction. I had a basic understanding of how the templating works, but I didn't realize the config file could be setup with `chezmoi init` using a template config like the ones you provided as a working example. Wow, chezmoi is pretty advanced, I really like it more and more.
